### ISSUE TITLE
Running tests when post PR 

### ISSUE THREAD

**Issue by dadiorchen** [Greenstand/treetracker-wallet-api#114]
In our CI, if we can run tests to reinforce the code would be great, the senerior is like: when devs post a PR, run the tests automatically, so the dev would find if posted code is fine, and it also saves reviewers time to find out problems, it's good for our code quality. 

Currently our test command would run 3 type of tests:
```
    "test": "npm run test-unit; npm run test-integration;npm run test-repository",
```

1. Unit tests; 
2. Integration tests, these tests need setting a postgres DB to run the code.
3. Repository test, to test the models visiting DB, separate it from the unit tests is just some trade-off of some problem concerning some tech issue.  

Is connecting to DB a trouble for this?

**Comment by deepwinter** [comment]
connecting to DB will be a challenge here, at best we must install a seed locally or we could have a remote db in Digital Ocean to reset to a seed with each run - but it's a problem if multiple PRs are created at once.

**Comment by dadiorchen** [comment]
In wallet API's integration test, the seed data is self-sufficient, all I need is a correct DB with those tables.

**Comment by dadiorchen** [comment]
Can we use Github Action to create a DB and destroy it when finished? I read this: https://docs.github.com/en/free-pro-team@latest/actions/guides/creating-postgresql-service-containers

**Comment by deepwinter** [comment]
@dadiorchen Thanks for that article about using the postgres service container, I had not seen that!  I have set up CI for new PRs that follows what you suggest.  The CI runs all three test suites when a new PR is opened, and should report general success/failure (reporting in more detail takes more work). 

 I think that use lint would also be good - I included this in the github workflow but commented it out because it reported very many errors.
